src/commons/application/ApplicationTypes.ts

```diff
@@ -230,16 +240,40 @@ const sourceSubLanguages: Language[] = [
 ];
 
 export const sourceLanguages: SALanguage[] = sourceSubLanguages.map(sublang => {
+  const { chapter, variant } = sublang;
+  const supportedFeatures: LanguageFeatures = sublang['supports'] ?? {};
+
+  // Enable Data Visualizer for Source Chapter 2 and above
+  supportedFeatures.dataVisualizer = chapter >= Chapter.SOURCE_2;
+
+  // Enable Subst Visualizer only for default Source 1 & 2
+  supportedFeatures.substVisualizer =
+    chapter <= Chapter.SOURCE_2 && (variant === Variant.DEFAULT || variant === Variant.NATIVE);
+
+  // Enable Env Visualizer for Source Chapter 3 and above
+  supportedFeatures.envVisualizer =
+    chapter >= Chapter.SOURCE_3 && variant !== Variant.CONCURRENT && variant !== Variant.NON_DET;
```

Comment:
I decided to backtrack on manually specifying the language configurations and placed them in a test file instead. I think going with this "rule-based" approach seems neater